Shared/Samples/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningView.Model.swift

```diff
@@ -0,0 +1,202 @@
+            
+            // Gets the pathways layer (optional for creating the IndoorsLocationDataSource).
+            let pathwaysLayer = map.operationalLayers.first(where: { $0.name == "Pathways" }) as! FeatureLayer
+            // Gets the levels layer (optional for creating the IndoorsLocationDataSource).
+            let levelsLayer = map.operationalLayers.first(where: { $0.name == "Levels" }) as! FeatureLayer
+            
+            // Setting up IndoorsLocationDataSource with positioning, pathways tables and positioning ID.
+            // positioningTable - the "IPS_Positioning" feature table from an IPS-aware map.
+            // pathwaysTable - An ArcGISFeatureTable that contains pathways as per the ArcGIS Indoors Information Model.
+            // Setting this property enables path snapping of locations provided by the IndoorsLocationDataSource.
+            // levelsTable - An ArcGISFeatureTable that contains floor levels in accordance with the ArcGIS Indoors Information Model.
+            // Providing this table enables the retrieval of a location's floor level ID.
+            // positioningID - an ID which identifies a specific row in the positioningTable that should be used for setting up IPS.
+            return IndoorsLocationDataSource(
+                positioningTable: positioningTable,
+                pathwaysTable: pathwaysLayer.featureTable as? ArcGISFeatureTable,
+                levelsTable: levelsLayer.featureTable as? ArcGISFeatureTable,
+                positioningID: positioningID
+            )
+        }
+        
+        /// The method that updates the location when the indoors location datasource is triggered.
+        func updateDisplayOnLocationChange() async throws {
+            for await location in locationDisplay.dataSource.locations {
+                // Since this listens for new location changes, it is important
+                // to ensure any blocking UI is dismissed once location updates begins.
+                // Floors in location are zero indexed however floorManager levels begin at one. Since
+                // it is necessary to display the same information to the user as the floor manager levelNumber
+                // one is added to the floor level value.
+                if let floorLevel = location.additionalSourceProperties[.floor] as? Int,
+                   (floorLevel + 1) != currentFloor {
+                    // Sets the currentFloor to the new floor level and adds one, since location uses
+                    // zero based flooring system.
+                    currentFloor = floorLevel + 1
+                    // The floor manager is used to filter so that only the data from the current floor is
+                    // displayed to the user.
+                    if let floorManager = map.floorManager {
+                        floorManager.levels.forEach {
+                            $0.isVisible = currentFloor == $0.levelNumber
```

Comment:
`Sequence.forEach(_:)` is intended to only be used at the end of a method chain. It shouldn't be used when a for loop would work instead.

```suggestion
                        for level in floorManager.levels {
                            level.isVisible = currentFloor == level.levelNumber
```